### ISSUE TITLE
RN App Crashes when opening synced Realm with Realm@v10.0.1

### ISSUE THREAD

**Issue by MadeByRaymond** [realm/realm-js#3409]
## Goals

Successfully open synced realms, read and write realm objects

## Expected Results

Open Realm with sync configurations.

## Actual Results

App crashes with no warnings/errors when `Realm.open(config)` is run .

![87794332-5c486f80-c83e-11ea-8be8-9970061e9d0c](https://user-images.githubusercontent.com/29353783/99191947-5b820480-2770-11eb-8281-28b328cc84a7.jpg)

**Seen on MongoDB Realm Log:**
`Logs:
[
  "Session closed after receiving UNBIND event",
  "Session was active for: 0s"
]
Partition:

<null>
Session Metrics:
{
  "downloads": 1
}
SDK:
react-native vRealmJS/10.0.1
Platform Version:
29` 

## Code Sample
_See project repo [here](https://github.com/MadeByRaymond/lysts-app.git). Pull dev branch_
`
const config = {
    schema: [MySchema],
    sync: {user: currentUser, partitionValue: null}
};

Realm.open(config).then(realm => {
    console.log("Successfully opened realm");
}).catch((e) => {
    console.log(e);
})
`

## Version of Realm and Tooling

- Realm JS SDK Version: v10.0.1
- Node or React Native: RN 0.62.2
- Client OS & Version: Android

**Comment by steffenagger** [comment]
First of all, we definitely have a bug here. The promise should reject with an error, _not_ crash the app.

To get you unblocked in the meanwhile, I can recommend Flipper. 
Through Flipper I got:
```sh
Ignored the notification for the crash 
{notificationID: "1", callstack: "/Volumes/Android/buildbot/src/android/ndk-release-r21/external/libcxx/../../external/libcxxabi/src/abort_message.cpp:72: abort_message: assertion "terminating with uncaught exception of type realm::InvalidAdditiveSchemaChangeException: The following changes cannot be made in additive-only schema mode:
- Property 'wishlist._id' has been made required.
- Property 'wishlist._partition' has been made required.
- Property 'wishlist.category' has been made required.
- Property 'wishlist.code' has been made required.
- Property 'wishlist.dateCreated' has been made required.
- Property 'wishlist.dateModified' has been made required.
- Property 'wishlist.description' has been made required.
- Property 'wishlist.listItems' has been changed from 'array<wishlist_listItems>' to 'array<items>'.
- Property 'wishlist.name' has been made required.
- Property 'wishlist.owner' has been made required.
- Property 'wishlist.status' has been made required." failed
Fatal signal 6 (SIGABRT), code -6 (SI_TKILL) in tid 2162 (mqt_js), pid 2083 (com.lysts)", name: "/Volumes/Android/buildbot/src/android/ndk-release-r21/external/libcxx/../../external/libcxxabi/src/abort_message.cpp:72: abort_message: assertion "terminating with uncaught exception of type realm::InvalidAdditiveSchemaChangeException: The following changes cannot be made in additive-only schema mode:", reason: "Cannot figure out the cause", date: Thu Nov 19 2020 08:12:29 GMT+0100 (Central European Standard Time)}
```

If the promise had rejected with this error, it should tell you that the schema was changed in a non-additive manner (all properties were changed to required, and the type `wishlist_listItems` changed name to `items` and from an embedded list to linked objects).

If you reinstate you previous schema, the app opens the realm:
```js
schema: [
    {
        name: 'wishlist',
        primaryKey: '_id',
        properties: {
        _id: 'objectId?',
        _partition: 'string?',
        category: 'string?',
        code: 'string?',
        dateCreated: 'date?',
        dateModified: 'date?',
        description: 'string?',
        listItems: 'wishlist_listItems[]',
        name: 'string?',
        owner: 'string?',
        status: 'string?',
        },
    },
    {
        name: 'wishlist_listItems',
        embedded: true,
        properties: {
        item: 'string?',
        status: 'string?',
        },
    },
]
```

I can only assume you changed the schema for a reason, and are not interested in going back. In this case you'll need to clear the schema/data serverside. Then the client will be allowed the changes.

**Comment by MadeByRaymond** [comment]
Thanks @steffenagger . This helped. 
However, I do still believe the app is not meant to crash and that should probably be looked at.

**Comment by steffenagger** [comment]
> However, I do still believe the app is not meant to crash and that should probably be looked at.

Yes, I've created an issue for the specific error, and will be closing this one.
Please see/track #3414 for progress.

**Comment by duncangroenewald** [comment]
@steffenagger - I too have this issue - BTW what is Flipper and how did you use it ?  Thanks

**Comment by kraenhansen** [comment]
@duncangroenewald please open your own issue (with a description that will allow us to reproduce it).
See https://fbflipper.com/docs/features/react-native/ for information on Flipper.

**Comment by duncangroenewald** [comment]
@kraenhansen have already done that.
